### ISSUE TITLE
Check that progress bar and hide_input work in IPython 3 and 4

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo#687]
I haven't checked this yet, but I suspect everything in `nengo/utils/ipython.py` needs to be updated for IPython 3. For the progress bar, [the widget migration guide](http://ipython.org/ipython-doc/3/whatsnew/version3_widget_migration.html) should be helpful.

**Comment by jgosmann** [comment]
The command line progress bar will show up instead of the IPython widget version. Furthermore, a warning is issued when importing Nengo:
```
/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/IPython/nbformat/current.py:19: UserWarning: IPython.nbformat.current is deprecated.

- use IPython.nbformat for read/write/validate public API
- use IPython.nbformat.vX directly to composing notebooks of a particular version

  """)
:0: FutureWarning: IPython widgets are experimental and may change in the future.
```

**Comment by jgosmann** [comment]
> If you distribute your custom widget, you may be using display and Javascript to publish the widget’s Javascript to the front-end. That is no longer the recommended way of distributing widget Javascript. Instead have the user install the Javascript to his/her nbextension directory or their profile’s static directory.

Doing it the recommended way will require user action, I suppose. Or can we do that somehow in setup.py?

**Comment by jgosmann** [comment]
Supposedly, the progress bar won't work after a page refresh in IPython 2. That's probably not really fixable.

**Comment by jgosmann** [comment]
* The progress bar works with IPython 3 without changes. A few changes might be in order to follow the recommended way of doing things (though, we have to keep a fall back to the current way for backwards compatibility for now).
* The detection of the IPython notebook does not work anymore and there seems to be no way of fixing this. We could users have to set some kind of global variable in notebooks to enable the fancy progress bar. That code line could go then in a profile file where it would automatically executed for notebooks. In case we're doing the recommended way of deploying the JS code and do that somehow with setup.py we might be able to add that line automatically.

**Comment by jgosmann** [comment]
WIP PR #693

**Comment by jgosmann** [comment]
IPython 4 was just released. So we should test with that too.

**Comment by tbekolay** [comment]
Yeah, I just saw that. What a pain! Their repos are all moved around; it's now `jupyter` everywhere, and there are top-level `nbconvert` modules, etc. :rage:

**Comment by jgosmann** [comment]
And they break everything again :rage: :boom:

**Comment by jgosmann** [comment]
Having trouble with sizing the progress bar widget correctly in IPython 4. Might be a bug on the site (if it's not a styling change done on purpose). See ipython/ipywidgets#143.

The only work around at the moment seems to be to set the progress bar to a fixed width or use JS to mess around with the notebook styling.

**Comment by jgosmann** [comment]
Fixed in PR #693.
